### PR TITLE
Allow add_points() to be called without an argument to create empty points layer

### PR DESCRIPTION
**Pull request by jni**
# Description
Similarly for add_shapes(). This allows easier annotation of volumes when working in an interactive IPython session.

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [x] example: the test suite for my feature covers all new code paths
- [x] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by jni** [comment]
@sofroniewn it would be awesome to get this added to 0.2.1 also...!

**Review by sofroniewn — APPROVED** [review]
Can you change the function signatures for `view_points` and `view_shapes` too?

How about doing the same for `add_labels` and `view_labels`?

File: napari/components/viewer_model.py [hunk 1]
```diff
@@ -973,18 +981,10 @@ def add_vectors(
         return layer
 
     def _new_points(self):
```
Comment (sofroniewn) [inline napari/components/viewer_model.py]: can we just drop the `_new_points` method entirely and connect the button to `self.add_points()` instead - one less function to worry about. similar for `_new_shapes` too

Comment (jni) [reply napari/components/viewer_model.py]: Yes but I worried about existing tests and having to replicate/modify them. It's almost 1am here. ;) I'd prefer to leave that for a later PR.

**Comment by jni** [comment]
Hmm looks like I have some fixing to do already anyway...!

**Comment by jni** [comment]
Re labels, it was a bit more complicated logic, which I thought I might want to tweak, so I decided to leave that for later... I can add it if you feel strongly about it.

Re `view_`, it seems to me a bit weird to launch an empty viewer + an empty points layer... What do you think?

**Comment by jni** [comment]
Ah, the class signature no longer matches the layer signature... HMMM. I guess I can add the `=None` to the layer, but it won't be quite the same, since there will be no way to know what the dimensionality of an empty layer should be. It'll have to be 2 by necessity.

**Comment by jni** [comment]
Hmmm, ok, I'm going to bed, as I don't know whether I can properly think about out whether it's prudent to remove Kira's very rigorous test :joy: If you figure it out, feel free to push to this branch and merge! I'll keep at it tomorrow. Goodnight!

**Comment by jni** [comment]
@sofroniewn how's the jet lag? ;) Anyway, this should work now! I'm pretty busy for the next couple of days (Lou Scheffer is actually in Melb! =), so I suggest merging this and leaving your other suggestions for a later release.

**Comment by jni** [comment]
... Aaaaand conda is broken. Anyway, tests pass locally...

**Comment by jni** [comment]
This is the cause of the conda failure:

https://github.com/conda/conda/pull/9014
https://github.com/conda/conda/issues/8149

Will attempt to solve with something similar to this:

https://github.com/umccr/woofr/commit/0b6b3633f095f8000302cc3eb60e98380a1dc33c

although maybe I can just delete our caches.

**Comment by jni** [comment]
Woot! :tada:

**Comment by sofroniewn** [comment]
I'd rather you can do `view_points()` and `view_shapes()` and have it add an empty layer too - I see no reason for the api to diverge there, and if people want to do it they then can.

I'll quickly make this change and maybe drop `_new_shapes` and `_new_points` is easy and then merge

**Comment by sofroniewn** [comment]
Changes made, will merge once tests pass

**Pull request merged**